src/ripple/app/misc/Transaction.h

```diff
@@ -154,6 +154,118 @@ class Transaction
+     * @brief Detect if transaction was accepted
+     * @param applied ref to store if transaction was appiled
+     * @param broadcast ref to store if transaction was appiled
+     * @param queued ref to store if transaction was appiled
+     * @param kept ref to store if transaction was appiled
+     * @return true if (applied|broadcast|queued|kept)
+     */
+    bool getAccepted(bool &applied, bool &broadcast,
+        bool &queued, bool &kept) const
+    {
+        applied = mApplied;
+        broadcast = mBroadcast;
+        queued = mQueued;
+        kept = mKept;
+        return mApplied || mBroadcast || mQueued || mKept;
+    }
+
+    /**
+     * Set this flag once was applied to open ledger.
+     */
+    void setApplied() {
+        mApplied = true;
+    }
+
+    /**
+     * Set this flag once was put into heldtxns queue.
+     */
+    void setQueued() {
+        mQueued = true;
+    }
+
+    /**
+     * Set this flag once was broadcasted via network.
+     */
+    void setBroadcast() {
+        mBroadcast = true;
+    }
+
+    /**
+     * Set this flag once was put to localtxns queue.
+     */
+    void setKept() {
+        mKept = true;
+    }
+
+    /**
+     * @brief clear all accepted-related flags
+     */
+    void clearAccepted() {
+        mApplied = false;
+        mBroadcast = false;
+        mQueued = false;
+        mKept = false;
+    }
+
+    /**
+     * Get last account sequence in current ledger and first available.
+     *
+     * @param available ref to save first available account sequence
+     * @return last account sequence in current ledger or early.
+     */
+    std::uint32_t getAccountSequence(std::uint32_t &available) const
+    {
+        available = mAccountSeqAvail;
+        return mAccountSeqNext;
+    }
+
+    /**
+     * Returns minimum fee required at current moment.
+     *
+     * @return minimum fee measured in drops.
+     */
+    XRPAmount getMinimumFeeRequired() const
+    {
+        return mMinFeeRequired;
+    }
+
+    /**
+     * @brief Sets minimum required fee and sequences for the transaction
+     * @param fee minimum fee required for the transaction
+     * @param accountSeq first valid account sequence in current ledger
+     * @param availableSeq first available sequence for the transaction
+     */
+    void setRequiredFeeAndSeq(XRPAmount fee,
+        std::uint32_t accountSeq, std::uint32_t availableSeq)
+    {
+        mMinFeeRequired = fee;
+        mAccountSeqNext = accountSeq;
+        mAccountSeqAvail = availableSeq;
+    }
+
+    /**
+     * Sets last known index of validated ledger.
+     *
+     * @param index number of last validated ledger
+     */
+    void setValidatedLedgerIndex(LedgerIndex index)
+    {
+        mValidatedLedger = index;
+    }
+
+    /**
+     * Returns last known index of validated ledger.
+     *
+     * @return last index of validated ledger.
+     */
+    LedgerIndex getValidatedLedgerIndex() const
+    {
+        return mValidatedLedger;
+    }
+
```

Comment:
I'd remove all the functions that set and get the auxiliary ledger information and replace them with two functions:
```
    struct CurrentLedgerState
    {
        CurrentLedgerState() = delete;

        CurrentLedgerState(
            LedgerIndex li,
            XRPAmount fee,
            std::uint32_t accSeqNext,
            std::uint32_t accSeqAvail)
            : validatedLedger{li}
            , minFeeRequired{fee}
            , accountSeqNext{accSeqNext}
            , accountSeqAvail{accSeqAvail}
        {
        }

        LedgerIndex validatedLedger;
        XRPAmount minFeeRequired;
        std::uint32_t accountSeqNext;
        std::uint32_t accountSeqAvail;
    };

    boost::optional<CurrentLedgerState>
    getCurrentLedgerState() const
    {
        return mCurrentLedgetState;
    }

    void
    setCurrentLedgerState(
        LedgerIndex validatedLedger,
        XRPAmount fee,
        std::uint32_t accountSeq,
        std::uint32_t availableSeq)
    {
        mCurrentLedgetState.emplace(
            validatedLedger, fee, accountSeq, availableSeq);
    }
```
Where `mCurrentLedgerState` is `boost::optional<CurrentLedgerState> mCurrentLedgetState;`

Reply:
Done.